### PR TITLE
Feat : Fuzzy span ruler

### PR DESCRIPTION
**Pull request by schmarion**

### CODE REVIEW

File: buzz_el/entity_matcher/entity_matcher.py [hunk 1]
```diff
@@ -60,24 +45,20 @@ def __init__(
...
-        self._fuzzy_matcher = None
-
         self.build_string_matcher()
-        # self.build_fuzzy_matcher()
+
+        self._fuzzy_matcher = None
+        if self.use_fuzzy:
+            self.build_fuzzy_matcher()
 
     def __call__(self, doc: Doc) -> Doc:
```
Comment (msesboue) [inline buzz_el/entity_matcher/entity_matcher.py:63]: If the fuzzy matcher is not None, I would just run the fuzzy matcher. The fuzzy matcher will always match at least what the string matcher does match. Isn't it?

**Review by msesboue — CHANGES_REQUESTED** [review]
I think I like it :)

I am not sure I see how we can set a threshold for the fuzzy matching though.

Also, some files are changed though they do not require update, e.g., `knowledge_graph.py` or `graph_loader.py`.

For the notebooks, I would suggest pushing the code without the outputs.

File: buzz_el/entity_matcher/entity_matcher.py [hunk 2]
```diff
@@ -32,23 +20,20 @@ class EntityMatcher:
...
     """
 
     def __init__(
         self,
         knowledge_graph: KnowledgeGraph,
         spacy_model: Language,
         ignore_case: Optional[bool] = True,
-        min_r: Optional[int] = None,
-        fuzzy_funct: Optional[str] = None,
+        use_fuzzy: Optional[bool] = False,
```
[the remark this answers is not in the document]
Comment (schmarion) [reply buzz_el/entity_matcher/entity_matcher.py:36]: You can do it by passing a config in the method `build_fuzzy_matcher()`, I try to reproduce what you have done with the SpanRuler. 
Do you want me to add an example in the notebook ?

File: buzz_el/entity_matcher/entity_matcher.py (hunk 1 above)
Comment (schmarion) [reply buzz_el/entity_matcher/entity_matcher.py:63]: Oh yes ok, I didn't understand that in what we said and what we wrote, but it suits me fine I'll make the change.

File: buzz_el/entity_matcher/entity_matcher.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (msesboue) [reply buzz_el/entity_matcher/entity_matcher.py:40]: Need to update the doc with the changes in the entity matcher initialiser.

**Review by msesboue — CHANGES_REQUESTED** [review]

File: buzz_el/entity_matcher/__init__.py [hunk 3]
```diff
@@ -1 +1,2 @@
-from .entity_matcher import EntityMatcher
\ No newline at end of file
+from .entity_matcher import EntityMatcher
+from .fuzzy_ruler import FuzzyRuler
```
[the remark this answers is not in the document]
Comment (schmarion) [reply buzz_el/entity_matcher/__init__.py:2]: Maybe people will use it just like this, for example if we make it a spaCy component ? Like the spaczz SpanRuler

**Pull request merged**